pkg/pillar/cmd/volumemgr/updatestatusCT.go

```diff
@@ -0,0 +1,333 @@
+}
+
+// lookForVerified handles the split between PersistImageStatus and
+// VerifyImageStatus. If it only finds the Persist it returns nil but
+// sets up a VerifyImageConfig.
+// Also returns changed=true if the VolumeStatus is changed
+func lookForVerifiedCT(ctx *volumemgrContext, status *types.ContentTreeStatus) (*types.VerifyImageStatus, bool) {
+	changed := false
+	vs := lookupVerifyImageStatus(ctx, status.ObjType, status.ContentSha256)
+	if vs == nil {
+		ps := lookupPersistImageStatus(ctx, status.ObjType, status.ContentSha256)
+		if ps == nil || ps.Expired {
+			log.Infof("Verify/PersistImageStatus for %s sha %s not found",
+				status.ContentID, status.ContentSha256)
+		} else {
+			log.Infof("lookForVerified: Found PersistImageStatus: %s based on ImageSha256 %s ContentID %s",
+				status.DisplayName, status.ContentSha256, status.ContentID)
+			if !status.HasPersistRef {
+				log.Infof("lookForVerified: Adding PersistImageStatus reference for ContentTreeStatus: %s", status.ContentSha256)
+				AddOrRefCountPersistImageStatus(ctx, ps.Name, ps.ObjType, ps.FileLocation, ps.ImageSha256, ps.Size)
+				status.HasPersistRef = true
+				changed = true
+			}
+			//Marking the ContentTreeStatus state as VERIFIED as we already have a PersistImageStatus for the content tree
+			if status.State != types.VERIFIED {
+				status.State = types.VERIFIED
+				status.Progress = 100
+				changed = true
+			}
+			if status.FileLocation != ps.FileLocation {
+				status.FileLocation = ps.FileLocation
+				log.Infof("lookForVerified: Update FileLocation for %s: %s",
+					status.Key(), status.FileLocation)
+				changed = true
+			}
+			// If we don't already have a RefCount add one
+			if !status.HasVerifierRef {
+				log.Infof("!HasVerifierRef")
+				// We don't need certs since Status already exists
+				MaybeAddVerifyImageConfigCT(ctx, *status, false)
+				status.HasVerifierRef = true
+				changed = true
+			}
+			//Wait for VerifyImageStatus to appear
+			return nil, changed
+		}
+	} else {
+		log.Infof("Found %s based on ContentID %s sha %s",
+			status.DisplayName, status.ContentID, status.ContentSha256)
+		// If we don't already have a RefCount add one
+		// No need to checkCerts since we have a VerifyImageStatus
+		c := kickVerifierCT(ctx, status, false)
+		if c {
+			changed = true
+		}
+		return vs, changed
+	}
+	return vs, changed
+}
+
+func updateContentTreeStatus(ctx *volumemgrContext, contentSha256 string, contentID uuid.UUID) {
+
+	log.Infof("updateContentTreeStatus(%s)", contentID)
+	found := false
+	pub := ctx.pubContentTreeStatus
+	items := pub.GetAll()
+	for _, st := range items {
+		status := st.(types.ContentTreeStatus)
+		if status.ContentSha256 == contentSha256 {
+			log.Infof("Found ContentTreeStatus %s", status.Key())
+			found = true
+			changed, _ := doUpdateCT(ctx, &status)
+			if changed {
+				publishContentTreeStatus(ctx, &status)
+			}
+		}
+	}
+	if !found {
+		log.Warnf("XXX updateContentTreeStatus(%s) NOT FOUND", contentID)
```

Comment:
There will be lots of these since you currently call this function for volumes and call the volume function for CTs.

Reply:
No, there will be no warning because we are looking for the Sha in the status while updating and sha will be present in both VolumeStatus and ContentTreeStatus.

Reply:
OK